.github/workflows/push.yml

```diff
@@ -25,15 +47,47 @@ jobs:
       - name: Build project
         run: ~/.elan/bin/lake -Kenv=dev build LeanAPAP
 
+      - uses: actions/cache@v3
+        name: Mathlib doc Cache
+        with:
+          path: build/doc/Mathlib
+          key: DocGen4-${{ hashFiles('lake-manifest.json') }}
+          restore-keys: |
+            DocGen4-
```

Comment:
CI takes around 26min because I have cached the Mathlib doc. If you change lake-manifest.json, CI would be much longer ( + 15min ) .